server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -487,32 +488,34 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
             return routing;
         }
         AliasOrIndex.Alias alias = (AliasOrIndex.Alias) result;
+
+        List<String> indexReferencesWithRouting = new ArrayList<>();
```

Comment:
can we separate this into a different change, so we can discuss it?

Reply:
I will try to split it out. I think it may require a few mini changes to make things pass in the meantime.